aggregating-profiler/src/main/java/ru/fix/aggregating/profiler/engine/AggregatingReporter.java

```diff
@@ -78,25 +79,27 @@ private ProfilerReport buildReportAndReset(Optional<String> tagName,
         Map<String, Long> indicators = profiler.getIndicators()
                 .entrySet()
                 .stream()
-                .filter(entry -> ! tagName.isPresent() || entry.getValue().hasTag(tagName.get(), tagValue.orElse(null)))
-                .collect(Collectors.toMap(
-                        e -> {
-                            String name = e.getKey();
-                            if (!name.endsWith(INDICATOR_SUFFIX)) {
-                                name = name.concat(INDICATOR_SUFFIX);
-                            }
-                            return name;
-                        },
-                        e -> {
-                            try {
-                                return e.getValue().getProvider().get();
-                            } catch (Exception ex) {
-                                log.error(ex.getMessage(), ex);
-                            }
-                            return null;
-                        }));
-
-        List<ProfiledCallReport> collect = new ArrayList<>();
+                .filter(entry -> ! tagName.isPresent()
+                        || entry.getValue().hasTag(tagName.get(), tagValue.orElse(null)))
+                .map(entry -> {
+                        String name = entry.getKey();
+                        if (!name.endsWith(INDICATOR_SUFFIX)) {
+                            name = name.concat(INDICATOR_SUFFIX);
+                        }
+                        try {
+                            return new SimpleEntry(name, entry.getValue().getProvider().get());
+                        } catch (Exception ex) {
+                            log.error(ex.getMessage(), ex);
```

Comment:
Could you please add context to error message. At least indicator name?  Seems like this is not first time I have to point out importance of context information in error logs.

Reply:
ok